cpp/src/binaryop/binaryop.cpp

```diff
@@ -144,34 +148,96 @@ void scalar_col_valid_mask_and(gdf_size_type& out_null_count,
 
 namespace jit {
 
+  const std::string hash = "prog_binop";
+
+  const std::vector<std::string> compiler_flags { "-std=c++14" };
+  const std::vector<std::string> headers_name
+        { "operation.h" , "traits.h", cudf_types_h };
+  
+  std::istream* headers_code(std::string filename, std::iostream& stream) {
+      if (filename == "operation.h") {
+          stream << code::operation;
+          return &stream;
+      }
+      if (filename == "traits.h") {
+          stream << code::traits;
+          return &stream;
+      }
+      return nullptr;
+  }
+
 void binary_operation(gdf_column* out, gdf_scalar* lhs, gdf_column* rhs,
                       gdf_binary_operator ope) {
-  Launcher()
-      .setKernelInst("kernel_v_s", ope, Operator::Type::Reverse, out, rhs, lhs)
-      .launch(out, rhs, lhs);
+  
+  cudf::jit::launcher(
+    hash, code::kernel, headers_name, compiler_flags, headers_code
+  ).set_kernel_inst(
+    "kernel_v_s", // name of the kernel we are launching
+    { cudf::jit::getTypeName(out->dtype), // list of template arguments
+      cudf::jit::getTypeName(rhs->dtype),
+      cudf::jit::getTypeName(lhs->dtype),
+      Operator{}.getOperatorName(ope, Operator::Type::Reverse) } 
+  ).launch(
+    out->size, out->data, rhs->data, lhs->data
+  );
+
 }
 
 void binary_operation(gdf_column* out, gdf_column* lhs, gdf_scalar* rhs,
                       gdf_binary_operator ope) {
-  Launcher()
-      .setKernelInst("kernel_v_s", ope, Operator::Type::Direct, out, lhs, rhs)
-      .launch(out, lhs, rhs);
+  
+  cudf::jit::launcher(
+    hash, code::kernel, headers_name, compiler_flags, headers_code
```

Comment:
Ideally we shouldn't have to add hashing to each functionality that utilizes Jitting.

Reply:
`hash` is probably not a good name here. This variable is intended to be the identifier/key of the cache.